src/addons/Radio/Radio.js

```diff
@@ -24,10 +27,22 @@ Radio.propTypes = {
     PropTypes.string,
     PropTypes.func,
   ]),
+
+  /** HTML input type, either checkbox or radio. */
+  inputType: Checkbox.propTypes.inputType,
+
+  /**
+   * Display as a checkbox, radio, slider, or toggle.
+   * The input type is `checkbox` for both slider and toggle types.
+   * You can set `inputType` separately to mix and match appearance and behavior.
+   */
+  type: Checkbox.propTypes.type,
 }
 
 Radio.defaultProps = {
   as: Checkbox,
+  inputType: 'radio',
```

Comment:
Why would a `Radio` ever not be an input type of `radio`? The overloading of `type`, where `inputType` is _actually_ the input `type`, seems pretty confusing. I could imagine a counterargument where users of `Radio` will rarely change the input type, which would mean `type` could be overloaded for other purposes (the styling of the input, as it currently is).

Still, just seems super confusing that `type` is no longer the actual type of the input.